src/algebra/group_with_zero_power.lean

```diff
@@ -0,0 +1,212 @@
+/-
+Copyright (c) 2020 Johan Commelin. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Johan Commelin
+-/
+
+import algebra.group_with_zero
+import algebra.group_power
+import algebra.field
+
+/-!
+# Powers of elements of groups with an adjoined zero element
+
+In this file we define integer power functions for groups with an adjoined zero element.
+This generalises the integer power function on a division ring.
+-/
+
+@[simp] lemma gwz.zero_pow {M : Type*} [monoid_with_zero M] :
+  ∀ n : ℕ, n ≠ 0 → (0 : M)^n = 0
+| 0     h := absurd rfl h
+| (k+1) h := zero_mul _
+
+namespace gwz
+variables {G : Type*} [group_with_zero G]
+
+section nat_pow
+
+@[simp, field_simps] theorem inv_pow (a : G) (n : ℕ) : (a⁻¹)^n = (a^n)⁻¹ :=
+by induction n with n ih; [exact inv_one.symm,
+  rw [pow_succ', pow_succ, ih, mul_inv_rev]]
+
+theorem pow_eq_zero {g : G} {n : ℕ} (H : g^n = 0) : g = 0 :=
+begin
+  induction n with n ih,
+  { rw pow_zero at H,
+    rw [← mul_one g, H, mul_zero] },
+  exact or.cases_on (mul_eq_zero _ _ H) id ih
+end
+
+@[field_simps] theorem pow_ne_zero {g : G} (n : ℕ) (h : g ≠ 0) : g ^ n ≠ 0 :=
+mt pow_eq_zero h
+
+theorem pow_sub (a : G) {m n : ℕ} (ha : a ≠ 0) (h : n ≤ m) : a^(m - n) = a^m * (a^n)⁻¹ :=
+have h1 : m - n + n = m, from nat.sub_add_cancel h,
+have h2 : a^(m - n) * a^n = a^m, by rw [←pow_add, h1],
+eq_mul_inv_of_mul_eq (pow_ne_zero _ ha) h2
+
+theorem pow_inv_comm (a : G) (m n : ℕ) : (a⁻¹)^m * a^n = a^n * (a⁻¹)^m :=
+by rw inv_pow; exact inv_comm_of_comm (pow_mul_comm _ _ _)
+
+end nat_pow
+
+end gwz
+
+section int_pow
+open gwz int
+variables {G : Type*} [group_with_zero G]
+
+/--
+The power operation in a group with zero.
+This extends `monoid.pow` to negative integers
+with the definition `a^(-n) = (a^n)⁻¹`.
+-/
+def fpow (a : G) : ℤ → G
+| (of_nat n) := a^n
+| -[1+n]     := (a^(nat.succ n))⁻¹
+
+@[priority 10] instance : has_pow G ℤ := ⟨fpow⟩
```

Comment:
It's at places like this where I wish the notation being used wasn't G. It just looks like you're defining powers on a group.

Reply:
I've changed the notation to G\0.